1.traverseLinkedList.js

```diff
@@ -20,7 +20,14 @@
 
 
 function traverse(list) {
-
+    if(!list) return [];
+    let output = []
+    let arr = [list]
+    while(arr.length){
+        let currentNode = arr.shift()
+        output.push(currentNode.val)
+    }
+        return output
 }
```

Comment:
We don't need the arr variable. We can traverse a linked list by updating the currentNode. 
currentNode = list 
while(currentNode)
    //stuff
currentNode = currentNode.next